File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) 
{
    int profit = 0;
    int min = prices[0];
    for(int i=1 ; i<pricesSize ; i++)
    {
        int cost = prices[i] - min;
        if(cost > profit)
        {
            profit = cost;
        }
        if(min>prices[i])
        {
            min = prices[i];
        }
    }
    return profit;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) 
    {
        int min = prices[0];
        int cost = 0;
        int profit = 0;
        for(int x:prices)
        {
            cost = x - min;
            if(profit<cost)
                profit = cost;
            if(x<min)
                min = x;    
        }
        return profit;
    }
}